src/electron/routing_service.ts

```diff
@@ -68,28 +80,42 @@ interface NetError extends Error {
 }
 
 // Abstracts IPC with OutlineService in order to configure routing.
-export class WindowsRoutingService implements RoutingService {
+export class RoutingService implements RoutingServiceInterface {
   private ipcConnection: net.Socket;
 
   // Asks OutlineService to configure all traffic, except that bound for the proxy server,
   // to route via routerIp.
-  configureRouting(routerIp: string, proxyIp: string, isAutoConnect = false): Promise<void> {
+  configureRouting(routerIp: string, proxyIp: string, isAutoConnect = false): Promise<string> {
     return this.sendRequest({
       action: RoutingServiceAction.CONFIGURE_ROUTING,
       parameters: {proxyIp, routerIp, isAutoConnect}
     });
   }
 
   // Restores the default system routes.
-  resetRouting(): Promise<void> {
+  resetRouting(): Promise<string> {
     return this.sendRequest({action: RoutingServiceAction.RESET_ROUTING, parameters: {}});
   }
 
+  // Returns the name of the device
+  getDeviceName(): Promise<string> {
+    return this.sendRequest({action: RoutingServiceAction.GET_DEVICE_NAME, parameters: {}});
+  }
+
   // Helper method to perform IPC with the Windows Service. Prompts the user for admin permissions
   // to start the service, in the event that it is not running.
-  private sendRequest(request: RoutingServiceRequest, retry = true): Promise<void> {
+  private sendRequest(request: RoutingServiceRequest, retry = true): Promise<string> {
     return new Promise((resolve, reject) => {
-      this.ipcConnection = net.createConnection(`${SERVICE_PIPE_PATH}${SERVICE_PIPE_NAME}`, () => {
+
+      let ipcName = '';
```

Comment:
No need to recompute this every time - looks like it can be a constant.

Reply:
Do we need to watch for unsupported platform?

Reply:
Hmm. Seems like this class shouldn't even get created on an unsupported platform.